pkg/kubelet/util/podip/podip.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podip
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"strings"
+
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	"k8s.io/kubernetes/pkg/kubelet/util/ioutils"
+)
+
+func getIpGetterCommand(interfaceName, addrType string) []string {
+	return []string{"ip", "-o", addrType, "addr", "show", "dev", interfaceName, "scope", "global"}
```

Comment:
This maybe should use `netlink.AddrList` like https://github.com/luxas/kubernetes/blob/5bdb377dcec9a2e9b42c81640186ae334bc6e46e/pkg/kubelet/container_bridge.go#L143 if possible.

I know it was like this before, but what about containers without the `ip` command?
(For example a `FROM scratch` container with a statically linked binary)